Clinical-Genomics-Lund/bonsai
# Added view for analyzing and classifying variants with IGV integration

This PR adds a view for analyzing and classifying variants called by TbProfiler and called SNV and SV variants.

This PR depends on this `bonsai-prp` PR, https://github.com/Clinical-Genomics-Lund/bonsai-prp/pull/33

## Changes

 - Added view for analyszing variants with filtering (/samples/<sample_id>/resistance/variants)
 - Added IGV genome browser integration to variant analysis view.
 - Bonsai support display of SV and SNV variants.
 - A user can classify variant as accepted or as rejected based and annotate why it was dissmissed.
 - A user can annotate that a variant yeilds resistance to additional anitbiotics
 - Placeholder Export to LIMBS button to the sidebar in the variants view


### How to setup and perform the tests

1. build a new container which includes the linked version of bonsai-prp
2. Make sure that you mount the following directories to the api container. Copy the contents of my directories (/data/bnf/dev/markus/bonsai_dev/volumes/api) to your folder.
```
      - "./volumes/api/reference_genomes:/tmp/reference_genomes"
      - "./volumes/api/annotations:/tmp/annotations"
      - "./volumes/api/sample_data:/tmp/samples"
```
4. load the attached test sample to Bonsai

### Expected outcome 

- Filtering variants should work
- Classifying variant should work
- Annotating additional resistance should work
- IGV integration should work

A short description of the expected outcome of the test above.

### Additional context
[test_mtuberculosis_1.json](https://github.com/Clinical-Genomics-Lund/bonsai/files/14509329/test_mtuberculosis_1.json)
